docs/pages/access-controls/sso/azuread.mdx

```diff
@@ -54,102 +60,116 @@ Before you get started you’ll need:
-    - {name: "http://schemas.microsoft.com/ws/2008/06/identity/claims/groups", value: "93b110...", roles: ["dev"]}
-  # Populate entity_descriptor with the federator data xml or set entity_descriptor_url to automatically load from the SSO
-  entity_descriptor: |
-    <federationmedata.xml contents>
-```
-
-(!docs/pages/includes/sso/idp-initiated.mdx!)
+  - name: http://schemas.microsoft.com/ws/2008/06/identity/claims/groups
+    roles:
+    - dev
+    value: 41c94563...
+  - name: http://schemas.microsoft.com/ws/2008/06/identity/claims/groups
+    roles:
+    - access
+    value: 8adac502...
+  audience: https://mytenant.teleport.sh/v1/webapi/saml/acs/ad
+  cert: ""
+  display: Microsoft
+  entity_descriptor: ""
+  entity_descriptor_url: https://login.microsoftonline.com/ff882432.../federationmetadata/2007-06/federationmetadata.xml?appid=b8d06e01...
+  issuer: ""
+  service_provider_issuer: https://mytenant.teleport.sh/v1/webapi/saml/acs/ad
+  sso: ""
+version: v2
 
-Replace the `acs` field with your Teleport address, update the group IDs in the `attributes_to_roles` field with the actual Azure AD group ID values, and insert the downloaded Federation Metadata XML into the `entity_descriptor` field.
+```
 
 Create the connector using `tctl`:
 
 ```code
-$ tctl create azure-connector.yaml
+$ tctl create -f azure-connector.yaml
 ```
 
-<Admonition
-  type="tip"
-  title="Automatic signing key pair"
->
-  Teleport will generate a signing key pair and update the SAML connector with the `signing_key_pair`
-  property.
+With the connector in place on the cluster, you can test it with `tctl`:
 
-  ![Sample Connector Transform](../../../img/azuread/azuread-12-sampleconnector.png)
-
-  Alternatively, you can generate your own keypair when creating a connector using `openssl`:
+```code
+$ cat azure-connector.yaml | tctl sso test
+```
 
-  ```code
-  $ openssl req  -nodes -new -x509  -keyout server.key -out server.cer
-  ```
-</Admonition>
+Your browser should open and log you in to the Teleport cluster through your
+Azure AD credentials. If there are any problems, the CLI output will help you
+debug the connector configuration.
 
 ## Create a new Teleport role
 
-Create a Teleport role resource that will use external username data from the Azure AD connector to determine which Linux logins to allow on a host.
+Create a Teleport role resource that will use external username data from the
+Azure AD connector to determine which Linux logins to allow on a host.
 
-Users with the following `dev` role are only allowed to log in to nodes with the `access: relaxed` Teleport label. They can log in as either `ubuntu` or a username that
-arrives via the Azure AD connector. Users with this role cannot obtain admin access to Teleport.
+Users with the following `dev` role are only allowed to log in to nodes with
+the `access: relaxed` Teleport label. They can log in as either `ubuntu` or a
+username that arrives via the Azure AD connector. Users with this role cannot
```

Comment:
```suggestion
username that is passed in from the Azure AD connector. Users with this role can't
```
"arrives" sounds a bit odd to me and "via" is more ambiguous (multiple possible meanings) than choosing a common English word. I prefer the conversational tone of contractions, but maybe you wanted to emphasize this point (you CANNOT do X).